src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -0,0 +1,413 @@
+   *
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions isNotTerminated() {
+    final Optional<Record<ProcessInstanceRecordValue>> processInstanceRecord =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withBpmnElementType(BpmnElementType.PROCESS)
+            .withIntent(ProcessInstanceIntent.ELEMENT_TERMINATED).stream()
+            .findFirst();
+
+    assertThat(processInstanceRecord.isPresent())
+        .withFailMessage("Process with key %s was terminated", actual.getProcessInstanceKey())
+        .isFalse();
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance has passed an element with a specific
+   * element id exactly one time.
+   *
+   * @param elementId The id of the element
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions hasPassedElement(final String elementId) {
+    return hasPassedElement(elementId, 1);
+  }
+
+  /**
+   * Verifies the expectation that the process instance has not passed an element with a specific
+   * element id.
+   *
+   * @param elementId The id of the element
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions hasNotPassedElement(final String elementId) {
+    return hasPassedElement(elementId, 0);
+  }
+
+  /**
+   * Verifies the expectation that the process instance has passed an element with a specific
+   * element id exactly N amount of times.
+   *
+   * @param elementId The id of the element
+   * @param times The amount of times the element should be passed
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions hasPassedElement(final String elementId, final int times) {
+    final long count =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey()).withElementId(elementId)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED).stream()
+            .count();
+
+    assertThat(count)
+        .withFailMessage("Expected element with id %s to be passed %s times", elementId, times)
+        .isEqualTo(times);
+
+    return this;
+  }
+
+  /**
+   * Verifies the expectation that the process instance has passed the given elements in order.
+   *
+   * @param elementIds The element ids
+   * @return this {@link ProcessInstanceAssertions}
+   */
+  public ProcessInstanceAssertions hasPassedElementInOrder(final String... elementIds) {
+    final List<String> foundElementRecords =
+        StreamFilter.processInstance(recordStreamSource.processInstanceRecords())
+            .withProcessInstanceKey(actual.getProcessInstanceKey()).withElementIdIn(elementIds)
+            .withIntent(ProcessInstanceIntent.ELEMENT_ACTIVATING).stream()
```

Comment:
:x: this is inconsistent with `hasPassedElement` in the former the filter is on `ELEMENT_COMPLETED` here it is `ELEMENT_ACTIVATING`. In my mind, `ELEMENT_COMPLETED` is the better indicator and should also be applied here.